tests/dns.rs

```diff
@@ -0,0 +1,90 @@
+use coinswap::{
+    maker::error::MakerError,
+    market::rpc::{RpcMsgReq, RpcMsgResp},
+    utill::{read_message, send_message},
+};
+use std::{
+    fs, io,
+    io::BufRead,
+    net::TcpStream,
+    path::PathBuf,
+    process::{Child, Command},
+    sync::mpsc::{self, Receiver, Sender},
+    thread,
+    time::Duration,
+};
+
+fn spawn_directoryd_thread(log_sender: Sender<String>) -> Child {
+    let mut directoryd_process = Command::new("cargo")
+        .args(["run", "--bin", "directoryd"])
```

Comment:
Instead of running via Cargo, a cleaner approach is to call the `./target/debug/directoryd` which is the binary we are interested in. 

Then pass in any arg only for the binary. If not keep args empty.